refactor(build): Build and test refactor

circlingthesun opened the pull request:
This PR upgrades the build process to use gulp instead of grunt. It allows future changes to make use the larger and more active gulp plugin ecosystem (relative to grunt). In addition to this I've upgraded the tests to run on Jasmine 2 and not use jQuery. Using jQuery in tests makes it hard to spot problems that only occur when one isn't using jQuery. I've also removed bower which seems to be falling out of favour and replaced it with JSPM. This allows one to use libraries on NPM, github, and elsewhere (with plugins). 

Other changes include:
- Removal of the `mm.foundation.tpls` module.  Templates are now included with the module that uses them. Templateless builds are still possible.
- Templates are now located in the same directory as the module code. This is more inline with various style guides and angular's move towards components
- The default gulp task starts karma in the background and runs tests as code changes. The demo page is also opened up and will update with changes
- Unused ng-doc files were removed

jbrowning commented:
Hey @circlingthesun, thanks for sending this in.

Overall, this looks great and I think this was a good first step to eventually fixing #280. I like including the templates in the module directories and the gulp build process. Not sure yet about JSPM at this point but I'll look into it.

Have you been able to make progress with the remaining components?

circlingthesun commented:
Hey @jbrowning, in this PR JSPM is only used during testing. It doesn't affect anything else. I thought about using SystemJS to do the build, but I havn't really investigated further. I'm keeping the build as it is atm.

As for #280, I've ported most of the components. The topbar js has been canned in f6 so I implemented the dropdown menu that can be used with the topbar css in f6 instead. Joyride/Tour is not in f6 but it looks like they plan to add it as a plugin later. Otherwise typeahead, interchange, and rating has not yet been ported. I might also write a module for abide and switch.

There are still a bunch of tests that need to be ported. Specifically for modal. I fixed a bunch of animation bugs in the original component that necessitate that tests be asynchronous. I've not had time to look into them.

Have a look at http://circlingthesun.github.io/angular-foundation-6/ to get a feel of how far things are along.
